nx/lib/nx.ex

```diff
@@ -687,9 +687,11 @@ defmodule Nx do
     names = Nx.Shape.named_axes!(opts[:names] || names!(tensor_or_shape), shape)
     type = Nx.Type.normalize!(opts[:type] || Nx.Type.infer(max - min))
 
-    unless Nx.Type.float?(type) or (Nx.Type.integer?(type) and is_integer(min) and is_integer(max)) do
-      raise ArgumentError, "random_uniform/3 expects compatible types, got: #{inspect(type)}" <>
-                           " in range #{inspect(min)} to #{inspect(max)}"
+    unless Nx.Type.float?(type) or
+           (Nx.Type.integer?(type) and is_integer(min) and is_integer(max)) do
```

Comment:
I believe this will also fail when `min` or `max` are outside the acceptable range for the integer type. What I would do is something like:

```elixir
min_max_type =
  type
  |> Nx.Type.merge_scalar(min)
  |> Nx.Type.merge_scalar(max)

unless Nx.Type.float?(type) or (Nx.Type.integer?(type) and type == min_max_type) do
...
end
```

Reply:
Didn't think of that, should be fine now